### PR TITLE
MAINT: add hotfix for wheel build fail, for context->gh PR #10861

### PR DESCRIPTION
**Pull request by rlucas7**
#### Reference issue
See the discussion at the end of PR #10861  for context. 

#### What does this implement/fix?
The merge of the `reuse` feature of `binned_statistic_dd()` broke the wheel build.
This PR would (maybe/hopefully) fix the intermittent failure. 

#### Additional information
The feature enhancement PR (e.g. #10861) added a unit test which is supposed to catch the edge case when 2 bin edges have a numerically 0.0 edge case. 
@rgommers noted that the change causes some intermittent errors in the wheel build.
The log of the build fail is here: 
https://travis-ci.org/MacPython/scipy-wheels/jobs/596997894
Let's see if the PR fixes.

### CODE REVIEW

File: scipy/stats/_binned_statistic.py [hunk 1]
```diff
@@ -686,11 +686,12 @@ def _bin_numbers(sample, nbin, edges, dedges):
...
-        # Find the rounding precision
-        try:
-            decimal = int(-np.log10(dedges[i].min())) + 6
-        except RuntimeWarning:
-            raise ValueError('The smallest edge difference is numerically 0.')
+        with np.errstate(divide='raise'):
+            # Find the rounding precision
+            try:
+                decimal = int(-np.log10(dedges[i].min())) + 6
+            except Exception:
```
Comment (tylerjereddy) [inline scipy/stats/_binned_statistic.py]: Catching a generic `Exception` is best avoided, as you likely know. No other choice here?

**Review by tylerjereddy — COMMENTED** [review]
I've attempted to test the fix for the wheels repo proposed here by pointing that repo to the commit hash on your fork--perhaps that will allow an assessment from: https://github.com/MacPython/scipy-wheels/pull/54

This may not be crucial for `1.3.2` as I note there, since the new feature/test is for 1.4.0, though it is nice to have the master branch of the wheels repo passing when you are working there.

File: scipy/stats/_binned_statistic.py (hunk 1 above)
Comment (larsoner) [reply scipy/stats/_binned_statistic.py]: Perhaps the same fix as here?

https://github.com/scipy/scipy/pull/10930/files#diff-9b2d6b68348fecf8d09990be924c77e9R690-R692

Comment (larsoner) [reply scipy/stats/_binned_statistic.py]: (well maybe that plus the `FloatingPointError` that NumPy might also raise)

**Comment by tylerjereddy** [comment]
The cross-linked wheels PR is looking promising now in terms of pointing to this commit hash and showing that wheels will build--full matrices will take quite some time to run though, so if the first iteration of this PR passes I won't likely try to repeat that again.

**Comment by rgommers** [comment]
this also needs a rebase now

**Comment by rlucas7** [comment]
> The cross-linked wheels PR is looking promising now in terms of pointing to this commit hash >and showing that wheels will build--full matrices will take quite some time to run though, so if the >first iteration of this PR passes I won't likely try to repeat that again.

Should I still update w/changes in review? 

Not sure how that will work, and if it would further break things, @tylerjereddy can you provide guidance here? If I make changes I'll add the tuple of exceptions and the rebase that @rgommers was requesting too but not sure if now is the best time for that with the cross-link setup and some builds on-going.

The config change/repoint in travis is a neat trick.

**Comment by tylerjereddy** [comment]
Just go ahead & address reviewer comments I'd say (thanks!). Maybe I'll point the cross-ref PR at your updated version, but I don't think we normally go that far anyway so we're already being careful enough I think.

**Comment by rgommers** [comment]
Alternative fix in gh-10989. Very similar, but perhaps that one is slightly cleaner?

File: scipy/stats/_binned_statistic.py (hunk 1 above)
Comment (rlucas7) [reply scipy/stats/_binned_statistic.py]: in the rebase I added another commit w/the tuple approach.

**Comment by rlucas7** [comment]
> this also needs a rebase now

done.

>Alternative fix in gh-10989. Very similar, but perhaps that one is slightly cleaner?
They are pretty similar, whichever you guys prefer is fine with me. I think the zero equality will run into issues as @ilayn mentioned in gh-10989, that could be worked around with an `_is_close()` type of method that would satisfy all the platforms for which SciPy builds. That way (I'd guess) has less overhead than the context manager approach. It is a bit different style-wise compared to the rest of the package. Like I said, whatever you guys prefer, hopefully they both fix the issue.

**Comment by tylerjereddy** [comment]
@rlucas7 There was a slight preference by the team for #10989 so I went ahead & merged that. Thanks for helping drive this forward--hopefully this wheels issue is resolved now.

If it isn't fixed we'll have to revisit & your input will be helpful of course. I'll close this for now.

**Comment by rlucas7** [comment]
Hi Tyler,
No worries. Glad the builds are fixed. 
Thanks for the update. 

Sincerely, 

-Lucas Roberts
